### PR TITLE
Added run_extension for an ImageLayer object

### PR DESCRIPTION
**Pull request by m1-key**
I have added a function execute_layer in executor.py
of extensions folder and also in cve_bin_tool/executor.py
and in scancode/executor.py

Solves Issue: #857
Work towards: #850

Signed-off-by: m1key <shubhamtiwari.tiwari84@gmail.com>

### CODE REVIEW

File: tern/analyze/passthrough.py [hunk 1]
```diff
@@ -127,3 +127,17 @@ def run_extension(image_obj, ext_string, redo=False):
...
+def run_extension_layer(image_layer, ext_string, redo=False):
+    '''Depending on what tool the user has chosen to extend with, load that
+    extension and run it'''
+    try:
+        mgr = driver.DriverManager(
+            namespace='tern.extensions',
+            name=ext_string,
+            invoke_on_load=True,
+        )
+        return mgr.driver.execute(image_layer, redo)
```
Comment (nishakm) [inline tern/analyze/passthrough.py:141]: This should be `execute_layer` -> the new functions you have added to the extensions.

File: tern/extensions/cve_bin_tool/executor.py [hunk 2]
```diff
@@ -33,3 +33,10 @@ def execute(self, image_obj, redo=False):
             passthrough.execute_and_pass(layer, command, True)
             # for now we just print the results for each layer
             print(layer.analyzed_output)
+
+    def execute_layer(self, image_layer, redo=False):
+        command = 'cve-bin-tool -x -u now'
+        logger.debug("Analyzing layer %s", image_layer.layer_index)
+        passthrough.execute_and_pass(image_layer, command, True)
+        # for now we just print the results for each image_layer
+        print(image_layer.analyzed_output)
```
Comment (nishakm) [inline tern/extensions/cve_bin_tool/executor.py:42]: Good!

**Review by nishakm — CHANGES_REQUESTED** [review]
Looks good to me except for one change required to make this work

**Comment by m1-key** [comment]
@nishakm Sorry, I accidently closed the PR. I have opened another PR #880 . Sorry for the inconvenience.

File: tern/extensions/cve_bin_tool/executor.py (hunk 2 above)
Comment (m1-key) [reply tern/extensions/cve_bin_tool/executor.py:42]: > Good!

Thanks :)

**Review by nishakm — APPROVED** [review]

**Comment by nishakm** [comment]
Looks good. Merging

**Pull request merged**